mmtk/src/active_plan.rs

```diff
@@ -5,7 +5,34 @@ use mmtk::util::opaque_pointer::*;
 use mmtk::vm::ActivePlan;
 use mmtk::Mutator;
 use mmtk::Plan;
-use std::sync::Mutex;
+use std::sync::{Mutex, MutexGuard};
+
+struct OpenJDKMutatorIterator<'a> {
+    _guard: MutexGuard<'a, ()>,
```

Comment:
Guarding the global iterator instance (`static MaybeUninit<JavaThreadIteratorWithHandle> jtiwh;`) in C++ is one way to adapt to the new API.  It is as correct as the previous code.

But given that OpenJDK provides a proper iterator type `JavaThreadIteratorWithHandle`, I think a more straightforward way to implement `OpenJDKMutatorIterator` is wrapping an instance of `JavaThreadIteratorWithHandle` (or a smart pointer to it) right into the `OpenJDKMutatorIterator` itself.

Reply:
Done. I removed the global iterator, and moved that to each iterator. `JavaThreadIteratorWithHandle` is a `StackObj`, thus cannot be allocated in the heap. I create an uninitialized instance in Rust, and pass that to C++ to let it initialize the instance.